### PR TITLE
UCT/IB/UD/MLX5: CQE size fixes.

### LINKED ISSUE #1019: CQ size error
if ud_mlx5 is enabled you get ab error:
```
[1475102411.803692] [softiron-3:22825:0]        ud_mlx5.c:655  UCX  ERROR TX CQE size (128) is not 64
```

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #1019

### CODE REVIEW

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -358,13 +358,42 @@ static ucs_status_t uct_ib_iface_init_lmc(uct_ib_iface_t *iface,
...
-    if ((inl > 32) || (UCS_SYS_CACHE_LINE_SIZE > 64)) {
-        ibv_exp_setenv(ibv_context, "MLX5_CQE_SIZE", "128", 1);
+    static const char *cqe_size_env_var = "MLX5_CQE_SIZE";
+    const char *cqe_size_env_value;
+    size_t cqe_size_min, cqe_size;
+    char cqe_size_buf[32];
+    int ret;
+
+    cqe_size_min       = (inl > 32) ? 128 : 64;
+    cqe_size_env_value = getenv(cqe_size_env_var);
```
Comment (miked-mellanox) [inline src/uct/ib/base/ib_iface.c]: why not ibv_exp_getenv?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2521/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1874/ for details (Mellanox internal link).

File: src/uct/ib/base/ib_iface.c (hunk 1 above)
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c]: there is no such function "ibv_exp_getenv"

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by yosefe** [comment]
@MattBBaker can you please post ORNL failure?

**Comment by shamisp** [comment]
👍

**Comment by yosefe** [comment]
@MattBBaker is it /dev/shm here as well?

**Comment by MattBBaker** [comment]
@yosefe No, this one is a big deal failure. A lot of tests failed and they were all the same in posted errors.
```
[----------] 2 tests from udrc/test_ucp_fence64
[ RUN      ] udrc/test_ucp_fence64.atomic_add_fadd/1
[1475395796.132051] [pod9:20848:0]       ib_iface.c:377  UCX  ERROR MLX5_CQE_SIZE is set to 64, but at least 128 is required (inl: 64)
/var/lib/jenkins/jobs/UCX/workspace/contrib/../test/gtest/ucp/ucp_test.cc:262: Failure
Error: Invalid parameter
[  FAILED  ] udrc/test_ucp_fence64.atomic_add_fadd/1, where GetParam() = \ud,\rc (24 ms)
```
If you want I can send you a full log.

**Comment by yosefe** [comment]
@MattBBaker do you have anything in your machine which sets the environment variable MLX5_CQE_SIZE to 64?

**Comment by yosefe** [comment]
nevermind, i know what the problem is.

**Comment by shamisp** [comment]
what ?

**Comment by yosefe** [comment]
setenv for send cq affected the logic for recv cq

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2536/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1890/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp pls take a look

**Pull request merged**